packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java

```diff
@@ -0,0 +1,167 @@
+   * @param resolutionPreset Platform agnostic enum containing resolution information.
+   * @param cameraId Camera identifier of the camera for which to configure the resolution.
+   */
+  public ResolutionFeature(
+      CameraProperties cameraProperties, ResolutionPreset resolutionPreset, int cameraId) {
+    super(cameraProperties);
+    this.currentSetting = resolutionPreset;
+    this.cameraId = cameraId;
+
+    configureResolution(resolutionPreset, cameraId);
+  }
+
+  /**
+   * Gets the best possible {@link android.media.CamcorderProfile} for the supplied {@link
+   * ResolutionPreset}.
+   *
+   * @param cameraId Camera identifier which indicates the device's camera for which to select a
+   *     {@link android.media.CamcorderProfile}.
+   * @param preset The {@link ResolutionPreset} for which is to be translated to a {@link
+   *     android.media.CamcorderProfile}.
+   * @return The best possible {@link android.media.CamcorderProfile} that matches the supplied
+   *     {@link ResolutionPreset}.
+   */
+  public static CamcorderProfile getBestAvailableCamcorderProfileForResolutionPreset(
+      int cameraId, ResolutionPreset preset) {
+
+    switch (preset) {
+        // All of these cases deliberately fall through to get the best available profile.
+      case max:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_HIGH)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_HIGH);
+        }
+      case ultraHigh:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_2160P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_2160P);
+        }
+      case veryHigh:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_1080P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_1080P);
+        }
+      case high:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_720P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_720P);
+        }
+      case medium:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_480P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_480P);
+        }
+      case low:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_QVGA)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_QVGA);
+        }
+      default:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_LOW)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_LOW);
+        } else {
+          throw new IllegalArgumentException(
+              "No capture session available for current capture session.");
+        }
+    }
+  }
+
+  @VisibleForTesting
+  static Size computeBestPreviewSize(int cameraId, ResolutionPreset preset) {
+    if (preset.ordinal() > ResolutionPreset.high.ordinal()) {
```

Comment:
It would be good to add a comment as to why this is done; IIRC there was a PR that did this originally, and other PRs that have tried to revert it without knowing why it was there.